File: backend/app/routes/health.py
"""Health check and service configuration status."""

from __future__ import annotations

from fastapi import APIRouter

from ..services import brevo, nvidia

router = APIRouter(tags=["health"])


@router.get("/health")
async def health() -> dict[str, object]:
    return {
        "status": "ok",
        "services": {
            "chat": "configured" if nvidia.is_configured() else "fallback",
            "email": "configured" if brevo.is_configured() else "unconfigured",
        },
    }

File: backend/src/routes/health.ts
/** Health check endpoint. */

import { Router } from 'express'
import * as brevo from '../services/brevo'
import * as nvidia from '../services/nvidia'

export const healthRouter = Router()

healthRouter.get('/health', (_req, res) => {
  res.json({
    status: 'ok',
    services: {
      chat: nvidia.isConfigured() ? 'configured' : 'fallback',
      email: brevo.isConfigured() ? 'configured' : 'unconfigured',
    },
    // Per-variable readiness (booleans only, no secret values) so a missing
    // env var can be pinpointed without dashboard access.
    emailConfig: brevo.configStatus(),
  })
})
